modules/workflow-workflowoperation/src/main/java/org/opencastproject/workflow/handler/workflow/DuplicateEventWorkflowOperationHandler.java

```diff
@@ -0,0 +1,467 @@
+    for (final MediaPackageElement e: sourcePubElements) {
+      try {
+        // We first have to copy the media package element into the workspace
+        final MediaPackageElement element = (MediaPackageElement) e.clone();
+        final File sourceFile = workspace.get(element.getURI());
+        final URI tmpUri = workspace.put(
+                destination.getIdentifier().toString(), element.getIdentifier().toString(),
+                FilenameUtils.getName(element.getURI().toString()),
+                new FileInputStream(sourceFile)
+        );
+        element.setIdentifier(null);
+        element.setURI(tmpUri);
+
+        // Now we can distribute it to the new media package
+        destination.add(element); // Element has to be added before it can be distributed
+        final Job j = distributionService.distribute(CHANNEL_ID, destination, element.getIdentifier());
+        final MediaPackageElement distributedElement = JobUtil.payloadAsMediaPackageElement(serviceRegistry).apply(j);
+        destination.remove(element);
+
+        updateTags(distributedElement, removeTags, addTags, overrideTags);
+
+        PublicationImpl.addElementToPublication(newPublication, distributedElement);
+      } catch (Exception exception) {
+        throw new WorkflowOperationException(exception);
+      }
+    }
+
+    // Using an altered copy of the source publication's URI is a bit hacky, but it works without knowing the URI pattern...
+    String publicationUri = sourcePublication.getURI().toString();
+    publicationUri = publicationUri.replace(source.getIdentifier().toString(), destination.getIdentifier().toString());
+    publicationUri = publicationUri.replace(sourcePublication.getIdentifier().toString(), newPublicationId);
+    newPublication.setURI(URI.create(publicationUri));
+    destination.add(newPublication);
+  }
+
+  private MediaPackage copyDublinCore(
+          MediaPackage source,
+          MediaPackageElement sourceDublinCore,
+          MediaPackage destination,
+          List<String> removeTags,
+          List<String> addTags,
+          List<String> overrideTags) throws WorkflowOperationException {
+    final DublinCoreCatalog destinationDublinCore = DublinCoreUtil.loadEpisodeDublinCore(workspace, source).get();
+    destinationDublinCore.setIdentifier(null);
+    destinationDublinCore.setURI(sourceDublinCore.getURI());
+    destinationDublinCore.set(DublinCore.PROPERTY_TITLE, destination.getTitle());
+    destinationDublinCore.setFlavor(sourceDublinCore.getFlavor());
+    for (String tag: sourceDublinCore.getTags()) {
+      destinationDublinCore.addTag(tag);
+    }
+    updateTags(destinationDublinCore, removeTags, addTags, overrideTags);
+    try (InputStream inputStream = IOUtils.toInputStream(destinationDublinCore.toXmlString(), "UTF-8")) {
+      destination = ingestService.addCatalog(inputStream, "dublincore.xml", MediaPackageElements.EPISODE, destination);
```

Comment:
I'm wondering why you use the ingest service to add a file to this media package and not just put the file into the ws/wfr add it directly? You add a dependency to another module for only this single line.

Reply:
That's a good point. I mostly try to avoid redundancy but in this case, these are so few lines that it makes sense to drop the dependency instead.

Can you say something about Workspace vs. WorkingFileRepository? They look very similar. Is it sufficient if I use the Workspace here? The IngestService uses WorkingFileDirectory, but so far, I only have the workspace as a dependency in the WOH.